src/test/datascience/debugger.vscode.test.ts

```diff
@@ -0,0 +1,160 @@
+
+        // Check that the debug adapter is created
+        const debugAdapter = debuggingManager.getDebugAdapter(doc);
+        assert.isOk<KernelDebugAdapter | undefined>(debugAdapter, 'DebugAdapter not started');
+
+        // Wait for the stoped event
+        let msg = await waitForEvent<DebugProtocol.StoppedEvent>('stopped', debugAdapter!);
+
+        // Check that we're stopped on the cell
+        const stack = await session!.customRequest('stackTrace', {
+            threadId: msg.body.threadId
+        });
+        assert.isTrue(stack.stackFrames.length > 0, 'has frames');
+        assert.equal(stack.stackFrames[0].source?.path!, cell.document.uri.toString(), 'Stopped at the worng path');
+
+        // continue to the next line
+        await commandManager.executeCommand(Commands.RunByLineContinue, cell);
+
+        // Wait for the stoped event
+        msg = await waitForEvent<DebugProtocol.StoppedEvent>('stopped', debugAdapter!);
+        // Wait for variable view to be updated
+        await sleep(1000);
+
+        // Check variable view
+        const coreVariableView = await variableViewProvider.activeVariableView;
+        // eslint-disable-next-line @typescript-eslint/no-explicit-any
+        const variableView = (coreVariableView as any) as ITestWebviewHost;
+        const htmlResult = await variableView?.getHTMLById('variable-view-main-panel');
+        const expectedVariables = [{ name: 'a', type: 'int', length: '', value: '1' }];
+        verifyViewVariables(expectedVariables, htmlResult);
+
+        // Stop run by line and check that the cell ran
+        await commandManager.executeCommand(Commands.RunByLineStop);
+        await sleep(1000);
+        assert.isTrue(getCellOutputs(cell).includes('1'));
+    });
+
+    test('Run by Line - Interrupt', async function () {
+        // set up
+        await insertCodeCell('a=1\na', { index: 0 });
+        const doc = vscodeNotebook.activeNotebookEditor?.document!;
+        const cell = doc.getCells()![0]!;
+
+        // Start Run by Line
+        await commandManager.executeCommand(Commands.RunByLine, cell);
+
+        // Check that a debugging session is created
+        const session = debuggingManager.getDebugSession(doc);
+        assert.isOk<DebugSession | undefined>(session, 'Session not started');
+
+        // Check that the debug adapter is created
+        const debugAdapter = debuggingManager.getDebugAdapter(doc);
+        assert.isOk<KernelDebugAdapter | undefined>(debugAdapter, 'DebugAdapter not started');
+
+        // Wait for the stoped event
+        await waitForEvent<DebugProtocol.StoppedEvent>('stopped', debugAdapter!);
+
+        // Interrupt kernel and check that the cell didn't finish running
+        await commandManager.executeCommand(Commands.InterruptKernel, { notebookEditor: { notebookUri: doc.uri } });
+        assert.isTrue(getCellOutputs(cell).includes('KeyboardInterrupt'));
```

Comment:
You might need to wait on the cell output to be updated?

Reply:
Or, it might always be done by the time the Interrupt command is done, if that code awaits on the output updates. You might not want to depend on that though. You can wait on the onDidUpdateCellOutputs event here too

Reply:
You can use the existing api we have waitForCondition (please see other tests)

@roblourens Fyi That's just using a settimeout, not the best,  back when this was written the events wouldn't fire, however can update that later,